### ISSUE TITLE
Edit a file from a commit *at the revision of the commit*

### ISSUE THREAD

**Issue by filmil** [jesseduffield/lazygit#3414]
**Is your feature request related to a problem? Please describe.**

I sometimes forget to run a formatter on my files before sending them to presubmit, causing presubmit to warn me about that.
Once I get the feedback from presubmit, I'm already a few commits in. Usually it's "just" adding a newline in a file. Going back to the
said revision just to add a newline is possible today with git (of course), but it's very laborious to do. It would save me tremendous
amount of time if I could navigate to the commit/file combo, and edit in place. The `e` command allows editing that file, but the
edit is at HEAD, which is not what I'd like.

**Describe the solution you'd like**
Allow me to modify the hunk in the commit/file by adding edits.

**Describe alternatives you've considered**
Manually doing the above.

It's something like: grow a branch from the commit I want to edit, check that commit out, do the edit, commit amend, rebase the remainder on top of this, being careful not to duplicate changes.
Remove the temporary branch.

Maybe this functionality exists, but I missed it?

**Additional context**
N/A

**Comment by stefanhaller** [comment]
> Maybe this functionality exists, but I missed it?

Yes it does, this is the kind of thing that lazygit is really good at. I do this several times every day, and it's not laborious at all.

There are actually at least two ways to do that:

1) Make your changes at HEAD, stage them, then go to the Commits panel and select the target commit, and press shift-A to amend the changes into that commit.

2) Select the target commit in the Commits panel, and press `e` there. This does an interactive rebase that stops at the selected commit. You can now make your changes, and hit shift-A in the Files panel to amend your changes into that commit. Then, `m enter` to continue the rebase.

Depending on circumstances, one may work better than the other.

**Comment by filmil** [comment]
> 1. Make your changes at HEAD, stage them, then go to the Commits panel and select the target commit, and press shift-A to amend the changes into that commit.

Note, you can't do that if the change I need is already present in subsequent changes.  There is no change at HEAD you can make in that case. So one would need to move that change from the appropriate hunk of the subsequent change to this hunk in this change. But I don't want to spelunk into all the subsequent commits to figure out where the change happened. I just want to make an edit and proceed with the rebase.

> 2. Select the target commit in the Commits panel, and press `e` there. This does an interactive rebase that stops at the selected commit. You can now make your changes, and hit shift-A in the Files panel to amend your changes into that commit. Then, `m enter` to continue the rebase.

This might work, tho. I'll try.
